src/main/java/org/testng/internal/ExitCode.java

```diff
@@ -0,0 +1,148 @@
+ * | 1                   | 1       | 1      | 1       |       | 15          | FailedWithinSuccess/Skipped/Failed/Success tests |
+ * |---------------------|---------|--------|---------|-------|-------------|--------------------------------------------------|
+ */
+public class ExitCode {
+    /**
+     * Represents all the statuses that can be possible.
+     */
+    public enum Status {
+        PASSED,
+        FAILED,
+        PASSED_FAILED,
+        SKIPPED,
+        SKIPPED_PASSED,
+        SKIPPED_FAILED,
+        SKIPPED_FAILED_PASSED,
+        FAILEDWITHINSUCCESS,
+        FAILEDWITHINSUCCESS_SUCCESS,
+        FAILEDWITHINSUCCESS_FAILED,
+        FAILEDWITHINSUCCESS_FAILED_SUCCESS,
+        FAILEDWITHINSUCCESS_SKIPPED,
+        FAILEDWITHINSUCCESS_SKIPPED_SUCCESS,
+        FAILEDWITHINSUCCESS_SKIPPED_FAILED,
+        FAILEDWITHINSUCCESS_SKIPPED_FAILED_SUCCESS;
+
+        static Status parse(int status) {
+            for (Status each : values()) {
+                if ((each.ordinal() + 1) == status) {
+                    return each;
+                }
+            }
+            return null;
+        }
+    }
+
+    public static boolean hasFailureWithinSuccessPercentage(int retCode) {
+        Status status = Status.parse(retCode);
+        if (status == null) {
+            return false;
+        }
+        switch (status) {
+            case FAILEDWITHINSUCCESS:
+            case FAILEDWITHINSUCCESS_SUCCESS:
+            case FAILEDWITHINSUCCESS_FAILED:
+            case FAILEDWITHINSUCCESS_FAILED_SUCCESS:
+            case FAILEDWITHINSUCCESS_SKIPPED:
+            case FAILEDWITHINSUCCESS_SKIPPED_SUCCESS:
+            case FAILEDWITHINSUCCESS_SKIPPED_FAILED:
+            case FAILEDWITHINSUCCESS_SKIPPED_FAILED_SUCCESS:
+                return true;
+            default :
+                return false;
+        }
+    }
+
+    public static boolean hasSkipped(int retCode) {
+        Status status = Status.parse(retCode);
+        if (status == null) {
+            return false;
+        }
+        switch (status) {
+            case SKIPPED:
+            case SKIPPED_PASSED:
+            case SKIPPED_FAILED:
+            case SKIPPED_FAILED_PASSED:
+            case FAILEDWITHINSUCCESS_SKIPPED:
+            case FAILEDWITHINSUCCESS_SKIPPED_SUCCESS:
+            case FAILEDWITHINSUCCESS_SKIPPED_FAILED:
+            case FAILEDWITHINSUCCESS_SKIPPED_FAILED_SUCCESS:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    public static boolean hasFailure(int retCode) {
+        Status status = Status.parse(retCode);
+        if (status == null) {
+            return false;
+        }
+        switch (status) {
+            case PASSED_FAILED:
+            case SKIPPED_FAILED:
+            case SKIPPED_FAILED_PASSED:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    private BitSet initial = new BitSet(4);
```

Comment:
could be final

Reply:
WDYT if `ExitCode` has status methods? It will be really easy to implement with the bitset and no need to have enum.